src/dyad/core/dyad_core.c

```diff
@@ -427,7 +432,7 @@ DYAD_CORE_FUNC_MODS dyad_rc_t dyad_get_data (const dyad_ctx_t* ctx,
         goto get_done;
     }
     DYAD_LOG_INFO (ctx, "Establish DTL connection with DYAD module");
-    rc = ctx->dtl_handle->establish_connection (ctx->dtl_handle, DYAD_COMM_RECV);
+    rc = ctx->dtl_handle->establish_connection (ctx->dtl_handle);
```

Comment:
This is the major change. With the new DTL interface, DYAD_COMM_RECV is set at initialization. So, it does not need to be passed here.